### PR TITLE
Make Tx from & to addresses checksummed

### PR DESCRIPTION
**Pull request by arjunu**
- Tx.to and Tx.from is now `EthereumAddressWithChecksum` instead of string
- updated `tslint`
- moved `isAddress`, `isChecksumAddress` to shared-utils
- made `makeEthereumAddressChecksummed` accept string
- defined and used `__DEV__`

### CODE REVIEW

File: packages/web/app/modules/investment-flow/sagas.ts [hunk 1]
```diff
@@ -444,8 +445,8 @@ function* setTransactionWithPresetGas(): any {
     actions.txSender.setTransactionData({
       gas: INVESTMENT_GAS_AMOUNT,
       value: "",
-      to: "",
-      from: "",
+      to: "" as EthereumAddressWithChecksum,
+      from: "" as EthereumAddressWithChecksum,
```
Comment (arjunu) [inline packages/web/app/modules/investment-flow/sagas.ts:449]: not clean, I'd say we rewrite this to not set empty values

File: packages/shared-utils/src/utils/address-utils.spec.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { expect } from "chai";
```
Comment (desfero) [inline packages/shared-utils/src/utils/address-utils.spec.ts:1]: i think we always use camelCase for file names so `addressUtils`

File: packages/shared-utils/src/utils/address-utils.spec.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    ];
+
+    addresses.forEach(address => {
+      expect(isAddress(address)).to.be.true;
+    });
+  });
+
+  it("should return false for an invalid address", () => {
+    const addresses = [
+      // mixed case
```
Comment (desfero) [inline packages/shared-utils/src/utils/address-utils.spec.ts:23]: mixed case comment should be applied to mixed case address not empty string

File: packages/shared-utils/src/utils/opaque-types/utils.ts [hunk 4]
```diff
@@ -16,7 +17,12 @@ const toEquityTokenSymbol = (symbol: string) => symbol as EquityToken;
  */
 const toEthereumAddress = (address: string) => address as EthereumAddress;
 
-const toEthereumChecksumAddress = (address: string) => address as EthereumAddressWithChecksum;
+const toEthereumChecksumAddress = (address: string) => {
+  if (process.env.NODE_ENV === "development" && !isChecksumAddress(address)) {
```
Comment (desfero) [inline packages/shared-utils/src/utils/opaque-types/utils.ts:21]: is it working on the wallet side? Is there a process.env.NODE_ENV?

File: packages/web/app/lib/web3/light-wallet/LightWallet.ts [hunk 5]
```diff
@@ -91,7 +92,7 @@ export class LightWallet implements IPersonalWallet {
 
       const encodedTxData: IRawTxData = {
         from: txData.from,
-        to: addHexPrefix(txData.to!),
+        to: makeEthereumAddressChecksummed(addHexPrefix(txData.to!) as EthereumAddress),
```
Comment (desfero) [inline packages/web/app/lib/web3/light-wallet/LightWallet.ts:95]: maybe we can avoid force casting here? `Either by typings `txData.to` properly or making makeEthereumAddressChecksummed to accept `string`?

File: packages/shared-utils/src/utils/address-utils.spec.ts (hunk 2 above)
Comment (arjunu) [reply packages/shared-utils/src/utils/address-utils.spec.ts:1]: Renamed

File: packages/shared-utils/src/utils/address-utils.spec.ts (hunk 3 above)
Comment (arjunu) [reply packages/shared-utils/src/utils/address-utils.spec.ts:23]: Fixed

File: packages/shared-utils/src/utils/opaque-types/utils.ts (hunk 4 above)
Comment (arjunu) [reply packages/shared-utils/src/utils/opaque-types/utils.ts:21]: No it won't. I thought of introducing __DEV__ to web in a separate PR but I ending up doing it here

File: packages/web/app/lib/web3/light-wallet/LightWallet.ts (hunk 5 above)
Comment (arjunu) [reply packages/web/app/lib/web3/light-wallet/LightWallet.ts]: Updated `makeEthereumAddressChecksummed` to accept string

File: packages/design-system/webpack/webpack.config.js [hunk 6]
```diff
@@ -1,17 +1,13 @@
...
 const { CleanWebpackPlugin } = require("clean-webpack-plugin");
-const OptimizeCSSAssetsPlugin = require("optimize-css-assets-webpack-plugin");
 const webpack = require("webpack");
 const CopyWebpackPlugin = require("copy-webpack-plugin");
 const { BundleAnalyzerPlugin } = require("webpack-bundle-analyzer");
 
 const paths = require("./paths");
 const { peerDependencies } = require("../package");
 
-const isCircleCI = () => process.env.CI === "true" && process.env.CIRCLECI === "true";
```
Comment (arjunu) [inline packages/design-system/webpack/webpack.config.js:13]: unused var

File: packages/web/test/setupTests.ts [hunk 7]
```diff
@@ -37,6 +37,7 @@ const { window } = new JSDOM(``, {
 });
 
 (global as IGlobalWithWindow).window = window;
+global.__DEV__ = true;
```
Comment (arjunu) [inline packages/web/test/setupTests.ts:40]: set it true for tests

**Comment by arjunu** [comment]
Build: https://appcenter.ms/orgs/Neufund/apps/Neufund-Wallet-MVP-iOS/distribute/releases/26

**Review by desfero — APPROVED** [review]

**Pull request merged**